spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -481,18 +481,18 @@ public void open() {
 
     System.setProperty("scala.repl.name.line", "line" + this.hashCode() + "$");
 
-    /* create scala repl */
-    this.interpreter = new SparkILoop(null, new PrintWriter(out));
+    synchronized (sharedInterpreterLock) {
+      /* create scala repl */
+      this.interpreter = new SparkILoop(null, new PrintWriter(out));
```

Comment:
should we check if this.interpreter != null if another thread has initialized it?